### ISSUE TITLE
Fluent Combo-box showing selected value instead of text on when selected value changed.

### ISSUE THREAD

**Issue by anujans** [microsoft/fast-blazor#328]
Hi Team,

Fluent combo box is showing selected value instead of text on selected value changed. 
We have referring https://brave-cliff-0c0c93310.azurestaticapps.net/Combobox  demo site. 

 In below example, selected text is "Three" and value is 3.

<img width="626" alt="Screenshot 2023-02-22 133037" src="https://user-images.githubusercontent.com/120552820/220558580-90016e35-d88f-4fd1-9801-1ca3194bce7b.png">

Kindly suggest through which site we can take reference for the same and other controls.

Thank you in advance!!

**Comment by vnbaaij** [comment]
Hi, 

There seems to be an issue when using the `<FluentCombobox>` in combination with the `Option<>` type. In retrospect, I think this can never work as expected in this situation unless it would be with an `Option<string>` where the value and text are equal. I'll look into that more

If you use the examples above this one on the page you mentioned, it works as expected. The data source for the component is in that case just a list of strings like so:
```
public List<string> Hits { get; } = new()
    {
        "Please Please Me",
        "With The Beatles",
        "A Hard Day's Night",
        "Beatles for Sale",
        "Help!",
        "Rubber Soul",
        "Revolver",
        "Sgt. Pepper's Lonely Hearts Club Band",
        "Magical Mystery Tour",
        "The Beatles",
        "Yellow Submarine",
        "Abbey Road",
        "Let It Be",
    };
```

> Kindly suggest through which site we can take reference for the same and other controls.
Not sure what you mean with this. The site you are referring to is the correct demo environment.

**Comment by vnbaaij** [comment]
The issue with the `<FluentCombobox>` icw `Option<T>` has been solved in V2.1.0 final (which will be released later this week). See https://brave-cliff-0c0c93310-259.centralus.azurestaticapps.net/Combobox for a demo of the latest version.
